docs/Design/numerical-types.md

```diff
@@ -0,0 +1,134 @@
+[https://cplusplus.com/reference/cinttypes/](https://cplusplus.com/reference/cinttypes/).
+
+```c++
+U32 value = 10;
+printf("My value: %" PRId32, value); // Uses string constant concatenation
+```
+
+## F´ Semantic Integer Types Design
+
+
+Typically, in flight software we try to use fixed-sized types wherever possible as this produces consistent results
+across platforms. However, when building components for use on multiple different architectures it is nice to be able
+to use semantic integer types whose exact widths are determined by the platform and project instead of needing
+to pick  a one-size-fits-all fixed with type. For example, representing a size as an unsigned 64-bit integer is logical
+on some systems that have 64-bit data widths, but will not compile on systems that do not define 64-bit integers at all
+(8-bit systems often lack this data size).
+
+Thus, for these semantic types the actual data size needs to be configured. This configuration needs to be available
+from two sources: a given project, and a given platform. Platforms need to set the idea size for these types, and
+projects need to be able to control these sizes when operating across multiple platforms.
+
+### Platform Configured Types
+
+Platform developers should define the following semantic types in the `StandardTypes.hpp` header provided alongside
+their CMake platform and toolchain files. Each must also define a compiler directive to indicate the type was defined.
+If the type is not specified as seen through the compiler directive the `DefaultTypes.hpp` header will fill in a default
+definition. Each also defines a format specifier for use with the `printf` family of functions. Additionally, each
+defines a pair of constants of the form `<type>_MIN` and `<type>_MAX` to define the minimum and maximum values.
+
+| Platform Semantic Type  | Compiler Directive                 | Default          | Format Specifier      | Notes                       | 
+|-------------------------|------------------------------------|------------------|-----------------------|-----------------------------|
+| PlatformIndexType       | PLATFORM_INDEX_TYPE_DEFINED        | PlatformIntType  | PRI_PlatformIndexType | Ports indices               | 
+| PlatformSizeType        | PLATFORM_SIZE_TYPE_DEFINED         | PlatformUIntType | PRI_PlatformSizeType  | Sizes                       |
+| PlatformPointerCastType | PLATFORM_POINTER_CAST_TYPE_DEFINED | PlatformIntType  | PRI_PointerCastType   | Pointers stored as integers |
+| PlatformAssertArgType   | PLATFORM_ASSERT_ARG_TYPE_DEFINED   | PlatformIntType  | PRI_AssertArgType     | Argument to FW_ASSERT       |
+| PlatformIntType         | PLATFORM_INT_TYPE_DEFINED          | int              | PRI_PlatformIntType   | Deprecated                  |
+| PlatformUIntType        | PLATFORM_UINT_TYPE_DEFINED         | unsigned int     | PRI_PlatformUIntType  | Deprecated                  |
```

Comment:
Should we add NATIVE_INT_TYPE and NATIVE_UINT_TYPE in this table? We can label them as deprecated. Or is PlatformIntType and PlatformUIntType the equivalent?

Reply:
These are the equivalents.  We could consider adding these to the second tables.